### ISSUE TITLE
Add "Related News Stories" section into collection page

### ISSUE THREAD

**Issue by dhmacs** [iFixit/react-commerce#14]
https://valkyrie.ifixit.com/wp-json/wp/v2/posts

**Comment by kwiens** [comment]
Oh! Why don't you just use Algolia for this? I'm being dumb.

**Comment by dhmacs** [comment]
@kwiens can you tell me which index contains the data?

**Comment by kwiens** [comment]
wp_searchable_posts

We don't have a tag for device yet, but we will. @Annika29

**Comment by dhmacs** [comment]
@Annika29 the code for this is ready. Do you have an ETA for the device tag? Otherwise I'm gonna merge this as is and open another PR later when the device tag is ready.

**Comment by Annika29** [comment]
@dhmacs this is done!

This should give you what you need: 

Endpoint Details:
endpoint: http://valkyrie.local/wp-json/wp/v2/posts/related
request type: post
request body:

{
	"tags": ["Right to Repair", "video"],
	"per_page": "5"
}

Notes :
`tags` are required, accepts an array
`per_page` is optional and defaults to `3` accepts numeric value
posts by tags are returned in `random` order
for empty tags `[]` will return random posts

**Comment by Annika29** [comment]
@dhmacs 

This should give you what you need, tags gave us the most control.

Endpoint Details:
endpoint: http://valkyrie.local/wp-json/wp/v2/posts/related
request type: post
request body:

{
	"tags": ["Right to Repair", "video"],
	"per_page": "5"
}

Notes :
`tags` are required, accepts an array
`per_page` is optional and defaults to `3` accepts numeric value
posts by tags are returned in `random` order
for empty tags `[]` will return random posts

**Comment by dhmacs** [comment]
@Annika29 I thought we were going to use the Algolia index and add the device tag to that index, right? Do you want me to drop current Algolia implementation and replace it with this API? (no problem doing this, but just making sure that we are on the same page)

**Comment by Annika29** [comment]
We did some research into different options (which I am now realizing would have been more relevant to post on this issue and not the corresponding ifixit issue) and using the api gave the most control over the content we would get. Sorry for not updating here earlier on that decision. Does that work for you?

<img width="574" alt="Screen Shot 2021-06-01 at 10 44 55 AM" src="https://user-images.githubusercontent.com/35045100/120368026-69d7de00-c2c6-11eb-8034-3691e8bbd2bb.png">

**Comment by dhmacs** [comment]
@Annika29 yes! I'll work to update the code 👍

**Comment by dhmacs** [comment]
@Annika29 If I try to make a request I get a 401 response. Do I need to add some auth headers?

<img width="1083" alt="Screen Shot 2021-06-04 at 18 01 30" src="https://user-images.githubusercontent.com/4640135/120830547-e684e980-c55e-11eb-9d70-1f0c10b80780.png">

**Comment by Annika29** [comment]
I'm guessing that's because we have basic auth on the live site and I think you need basic auth for external access. Let me take a look now

**Comment by dhmacs** [comment]
@Annika29 Posting here a few things that seems to be missing from the API payload:

- Post category
- Post header image

Moreover, should I use the `guid` field as the link for the post?

**Comment by Annika29** [comment]
Thanks @dhmacs! We have an issue on our side to add the additional info needed which should be completed by this week. I'll update again here when I have this unblocked for you.

**Comment by Annika29** [comment]
sorry for the hold-up on this @dhmacs! This is actively being worked on!

@kwiens ref issue on our side: https://github.com/iFixit/valkyrie/issues/485

**Comment by Annika29** [comment]
This is soooo close to being unblocked for you, @dhmacs. This [pull](https://github.com/iFixit/valkyrie/pull/502) has been merged but needs to be deployed and that will be done tomorrow morning by @addison-grant, after which you will have all of the info you need (category, featured image) to move forward.

**Comment by Annika29** [comment]
@dhmacs this is unblocked and good to go!

**Comment by dhmacs** [comment]
Hey @Annika29, I'm having CORS issues when I make requests from localhost. Do you have a dev endpoint that allows requests from localhost? I tried cominor using the same api path, but it doesn't work either.

<img width="696" alt="Screen Shot 2021-07-05 at 20 01 20" src="https://user-images.githubusercontent.com/4640135/124506379-c6e61880-ddcb-11eb-8fb2-cbb1b4c619b7.png">

**Comment by Annika29** [comment]
Ah interesting issue, my guess would be this didn't come up in testing because it's fine via curl but this happens in the browser. @addison-grant does that sound right? I don't know if we have a dev endpoint that allows requests from localhost. That would solve your issues or we add CORS support.

**Comment by danielbeardsley** [comment]
I'm pretty sure we've only ever queried the wordpress API from within our app via a backend request. So accessing the api from the client via a different origin is new (unless I'm behind the times). Regarding access from localhost, we could add some CORS headers for that just in development.

Also, I see above that @Annika29 said "We solved this, basic auth acquired". Perhaps she meant "we've removed basic auth"? Otherwise, you'd have to have the user/pass embedded in the client code and I don't think we're going to do that.

**Comment by Annika29** [comment]
That checks out, yes. To move this forward, adding some CORS headers sounds reasonable and that would unblock this. I don't know if we will need to access the api from the client side from a different origin more in the future but that sounds like it requires more dev work, is that correct?

**Comment by dhmacs** [comment]
@Annika29 @danielbeardsley I can actually proxy the requests via a Next.js API route if it's only meant to be accessed this way.. The inconvenience is that we add an additional overhead to the request

**Comment by dhmacs** [comment]
I thought a little bit more about it, and I think that since we're doing SSR I can just fetch that data only server side so you don't have to touch anything on your side

**Comment by dhmacs** [comment]
@Annika29 I think @kwiens mentioned that we are going to embed the React collection into the Shopify theme before moving to a Next.js solution later this year. Taking this into account I think it is necessary to enable CORS on the API since we can only get posts data client side when we are on the Shopify theme

**Comment by Annika29** [comment]
Hey @dhmacs just confirming that this is unblocked for you? Happy to help but I think this is good to go, apologies there is an issue for this on the ifixit side and the communication doesn't always make it onto both issues.

**Comment by dhmacs** [comment]
Hey @Annika29 yes I'm diving back on this asap, I was busy fixing another issue this week. I'll try making some API requests, and I'll let you know if there are issues.

I've also moved this to the todo column in the project.. When you unblock a task feel free to move it back to the todo column 👍
